openml/flows/functions.py

```diff
@@ -1,13 +1,75 @@
 import dateutil.parser
-
+from collections import OrderedDict
+import os
+import io
+import re
 import xmltodict
+from typing import Union, Dict
+from oslo_concurrency import lockutils
 
+from ..exceptions import OpenMLCacheException
 import openml._api_calls
 from . import OpenMLFlow
 import openml.utils
 
+FLOWS_CACHE_DIR_NAME = 'flows'
+
+
+def _get_cached_flows() -> OrderedDict:
+    """Return all the cached flows.
+
+    Returns
+    -------
+    flows : OrderedDict
+        Dictionary with flows. Each flow is an instance of OpenMLFlow.
+    """
+    flows = OrderedDict()
+
+    flow_cache_dir = openml.utils._create_cache_directory(FLOWS_CACHE_DIR_NAME)
+    directory_content = os.listdir(flow_cache_dir)
+    directory_content.sort()
+    # Find all flow ids for which we have downloaded
+    # the flow description
+
+    for filename in directory_content:
+        if not re.match(r"[0-9]*", filename):
+            continue
+
+        fid = int(filename)
+        flows[fid] = _get_cached_flow(fid)
+
+    return flows
+
+
+def _get_cached_flow(fid: int) -> OpenMLFlow:
+    """Get the cached flow with the given id.
+
+    Parameters
+    ----------
+    fid : int
+        Flow id.
+
+    Returns
+    -------
+    OpenMLFlow.
+    """
+
+    fid_cache_dir = openml.utils._create_cache_directory_for_id(
+        FLOWS_CACHE_DIR_NAME,
+        fid
+    )
+    flow_file = os.path.join(fid_cache_dir, "flow.xml")
+
+    try:
+        with io.open(flow_file, encoding='utf8') as fh:
+            return _create_flow_from_xml(fh.read())
+    except (OSError, IOError):
+        openml.utils._remove_cache_dir_for_id(FLOWS_CACHE_DIR_NAME, fid_cache_dir)
+        raise OpenMLCacheException("Flow file for fid %d not "
+                                   "cached" % fid)
+
 
-def get_flow(flow_id, reinstantiate=False):
+def get_flow(flow_id: int, reinstantiate: bool = False) -> OpenMLFlow:
     """Download the OpenML flow for a given flow ID.
 
     Parameters
```

Comment:
you have forgotten to finish the sentence for the `reinstantiate` parameter.

Reply:
Unfortunately this is an artefact that exists in [develop](https://github.com/openml/openml-python/blob/develop/openml/flows/functions.py#L18). I am not sure what this extra clause would be so I can not complete it.